### PR TITLE
Add NEAT optimizer section to ecosystem.md

### PR DESCRIPTION
**Pull request by ItCodinTime**
Added NEAT optimizer documentation with features and usage example.

### CODE REVIEW

**Comment by ItCodinTime** [comment]
Was given permission by @hertschuh in this thread

https://github.com/keras-team/keras/issues/22639#issuecomment-4577513912

File: templates/getting_started/ecosystem.md [hunk 1]
```diff
@@ -64,3 +64,31 @@ A Python library for amortized Bayesian workflows using generative neural networ
...
+- Player-aware training utilities
+- Training diagnostics for conflict and update alignment
+
+```python
+import keras
+from neat_optim import NEAT
+
+model.compile(
+optimizer=NEAT(learning_rate=1e-3),
+loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
```
Comment (hertschuh) [inline templates/getting_started/ecosystem.md]: Indent these 2 lines

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for adding this!

File: templates/getting_started/ecosystem.md (hunk 1 above)
Comment (ItCodinTime) [reply templates/getting_started/ecosystem.md]: Will do! Ty for the reply!

**Review by ItCodinTime — COMMENTED** [review]
Indenting

**Review by hertschuh — APPROVED** [review]
Thanks!
